kfs-ld/src/main/java/edu/arizona/kfs/module/ld/batch/dataaccess/impl/LaborEncumbranceAdjustmentDaoJdbc.java

```diff
@@ -0,0 +1,148 @@
+                                                    "    HAVING SUM(l.ACLN_ANNL_BAL_AMT) != 0\n" + 
+                                                    "    ORDER BY FIN_BALANCE_TYP_CD, EMPLID, position_nbr, fin_coa_cd, account_nbr, sub_acct_nbr," + 
+                                                    "             FIN_OBJECT_CD, FIN_SUB_OBJ_CD, fin_obj_typ_cd";
+    
+    public int buildFileForEncumbranceBalances(Integer fiscalYear, File fileName) {
+        // get the connection and use resultsets directly
+        PreparedStatement stmt = null;
+        ResultSet rs = null;
+        PrintWriter writer = null;
+        int outputLines = 0;
+        StringBuilder sb = new StringBuilder();
+        Map<String, Integer> lMap = new LaborOriginEntryFieldUtil().getFieldLengthMap();
+        // this is a generated tmp file, so if one exists, remove it
+        if (fileName.exists()) {
+            fileName.delete();
+        }
+        // get the fiscal year's internal encumbrance value
+        String internalEncumbranceBalanceTypeCode = SpringContext.getBean(OptionsService.class).getOptions(fiscalYear).getIntrnlEncumFinBalanceTypCd();
+
+        String preencumbranceFinBalTypeCd;
+        // **START AZ** KFSI-5984 KITT-3023 KevinMcO
+        preencumbranceFinBalTypeCd = SpringContext.getBean(OptionsService.class).getOptions(fiscalYear).getPreencumbranceFinBalTypeCd();
+        // **END AZ**
+
+        try {
+            // create an empty file, so that if no records are found, a file is still created (makes later processing easier)
+            fileName.createNewFile();
+            try {
+                stmt = getConnection().prepareStatement(ENCUMBRANCE_BALANCE_SQL + additionalWhereClause + ENCUMBRANCE_BALANCE_SQL2);
+                stmt.setInt(1, fiscalYear);
+                stmt.setString(2, internalEncumbranceBalanceTypeCode);
+                // **START AZ** KFSI-5984 KITT-3023 KevinMcO
+                stmt.setString(3, preencumbranceFinBalTypeCd);
+
+                rs = stmt.executeQuery();
+            }
+            catch (SQLException ex) {
+                LOG.error("Unable to create and execute statement to retrieve labor encumbrance balances", ex);
+                throw new RuntimeException("Unable to create and execute statement to retrieve labor encumbrance balances", ex);
+            }
+            if (LOG.isInfoEnabled()) {
+                LOG.info("Opening " + fileName.getAbsolutePath() + " for output of balance records.");
+                if (!StringUtils.isBlank(additionalWhereClause)) {
+                    LOG.info("Additional WHERE clause in use: " + additionalWhereClause);
+                }
+            }
+            writer = new PrintWriter(new BufferedWriter(new FileWriter(fileName)));
+            try {
+                StringBuffer lineBuffer = new StringBuffer(100);
```

Comment:
Suggestion: since we have our own stack frame here, the non-synchronized StringBuilder would be more performant here.